monitoring/uss_qualifier/suites/faa/uft/design/README.md

```diff
@@ -0,0 +1,79 @@
+# UFT Message Signing tests for SCD using Uss_Qualifier
+
+UTM Field Test (UFT) has a requirement for USSes to sign their message requests
+and responses in the SCD flow using IATF certificates provided by FAA.
+
+This test suite helps to test the message signing by extending [suites.astm.utm.f3548_21](../../../astm/utm/f3548_21.yaml)
+with scenarios that trigger reporting of message signing in the interactions with mock_uss.
+
+## Test Harness Setup
+
+The test setup includes running the following components on your local machine, as also shown in the [diagram](./InterUss%20Test%20Harness%20With%20Message%20Signing.png)
```

Comment:
Sure. We can work on this. As I understand, scd.yaml interface is already implemented and baked into mock_uss. We can take out the start/stop endpoints as a separate interface in mock_uss. 
If you would want to integrate the message signing logs into report.json, do you mean the message_signing logs will be POSTed to uss_qualifier, which means uss_qualifier would need an interface for that? Or the other way is after test run, the uss_qualifier GETs the message signing logs and has a way to associate each of the message signing logs to a specific uss_qulaifier test? That would mean more work on the mock_uss interface design. Which of the above ways are you thinking it should be?  Thanks.

Reply:
> As I understand, scd.yaml interface is already implemented and baked into mock_uss

Yes, mock_uss already implements scd.yaml -- it's valuable to clarify that "Inject operations" goes through that interface to clarify that uss_qualifier doesn't need to know that it's talking to a mock_uss instance in order to inject operations.  uss_qualifier *does* need to know that it's talking to a mock_uss when starting and stopping message signing logging since most USSs won't have that capability.

> after test run, the uss_qualifier GETs the message signing logs

I think that approach will work well (to be more specific, "[after the F3548-21 test suite](https://github.com/interuss/dss/blob/939110bb448665ced9bdcbf1161daba74ae4d13a/monitoring/uss_qualifier/suites/faa/uft/message_signing.yaml#L25)").

> uss_qualifier [...] has a way to associate each of the message signing logs to a specific uss_qulaifier test

uss_qualifier already has this -- [the message signing test suite](https://github.com/interuss/dss/blob/master/monitoring/uss_qualifier/suites/faa/uft/message_signing.yaml) performs three actions: [start message signing logging](https://github.com/interuss/dss/blob/fb41083a0e8d1185e5b0706c9ac8b28a814143fa/monitoring/uss_qualifier/scenarios/faa/uft/message_signing_start.py#L16), [F3548-21](https://github.com/interuss/dss/blob/939110bb448665ced9bdcbf1161daba74ae4d13a/monitoring/uss_qualifier/suites/faa/uft/message_signing.yaml#L16), [stop message signing logging](https://github.com/interuss/dss/blob/fb41083a0e8d1185e5b0706c9ac8b28a814143fa/monitoring/uss_qualifier/scenarios/faa/uft/message_signing_finalize.py#L16).  When uss_qualifier runs that last scenario (where it would obtain the logs from mock_uss under the proposed architecture), it knows what test it's on.  Logging a check from within a test scenario automatically attaches this association without any effort required from the developer.  Any general dict-like metadata can [already be attached to a FailedCheck](https://github.com/interuss/dss/blob/939110bb448665ced9bdcbf1161daba74ae4d13a/monitoring/uss_qualifier/reports/report.py#L47).  We could also add the ability to attach the same kind of open-ended data to a [PassedCheck](https://github.com/interuss/dss/blob/939110bb448665ced9bdcbf1161daba74ae4d13a/monitoring/uss_qualifier/reports/report.py#L51) if needed.

> That would mean more work on the mock_uss interface design.

We're already adding endpoints to mock_uss for starting and stopping message signing; this would either be a third endpoint like those, or else some additional data on the second endpoint.  These endpoints probably don't need to be put in YAML since we don't have any YAML for mock_uss endpoints currently because it's just a mock not intended for use outside InterUSS tooling -- YAML is useful to clearly define interfaces for external users of that interface, but we don't have external users here.